cryptoconditions/types/zenroom.py

```diff
@@ -0,0 +1,303 @@
+    # only time we need this (e.g. we could produce a bitcoin signed transaction and
+    # the code would be different)
+    def sign(self, message, condition_script, private_keys):
+        message = json.loads(message)
+        data = {}
+        if 'data' in message['asset'].keys():
+            data['asset'] = message['asset']['data']
+        if self.data is not None:
+            data['output'] = self.data
+        
+        result = ZenroomSha256.run_zenroom(condition_script,
+                                           {"keys": private_keys},
+                                           data)
+        message['metadata'] = {'data': json.loads(result.output),
+                               'result': 'ok'}
+
+        print(message)
+        return json.dumps(message)
+
+    # TODO Adapt according to outcomes of
+    # https://github.com/rfcs/crypto-conditions/issues/16
+    def to_json(self):
+        """
+        Generate a dict of the fulfillment
+
+        Returns:
+            dict: representing the fulfillment
+        """
+        return {
+            'type': ZenroomSha256.TYPE_NAME,
+            'script': base64_remove_padding(
+                urlsafe_b64encode(self.script)),
+            'keys': base64_remove_padding(
+                urlsafe_b64encode(self.keys)),
+            # 'conf': base64_remove_padding(
+            #     urlsafe_b64encode(self.conf)),
+        }
+
+    # TODO Adapt according to outcomes of
+    # https://github.com/rfcs/crypto-conditions/issues/16
+    def parse_dict(self, data):
+        """
+        Generate fulfillment payload from a dict
+
+        Args:
+            data (dict): description of the fulfillment
+
+        Returns:
+            Fulfillment
+        """
+        if data.get('script'):
+            self.script = base58.b58decode(data['script'])
+        if data.get('keys'):
+            self.keys = base58.b58decode(data['keys'])
+
+    # TODO Adapt according to outcomes of
+    # https://github.com/rfcs/crypto-conditions/issues/16
+    def parse_json(self, data):
+        """
+        Generate fulfillment payload from a dict
+
+        Args:
+            data (dict): description of the fulfillment
+
+        Returns:
+            Fulfillment
+        """
+        self.script = urlsafe_b64decode(base64_add_padding(
+            data['script']))
+        self.keys = urlsafe_b64decode(base64_add_padding(
+            data['keys']))
+
+    def parse_asn1_dict_payload(self, data):
+        self.script = data['script']
+        self.keys = data['keys']
+
+    def validate(self, *, message):
+        """
+        Verify the signature of this Zenroom fulfillment.
+
+        The signature of this Zenroom fulfillment is verified against
+        the provided message and script.
+
+        Args:
+            message (str): Message to validate against.
+
+        Return:
+            boolean: Whether this fulfillment is valid.
+        """
+        message = json.loads(message)
```

Comment:
could you also catch the loading exceptions here, just in case something unexpected gets entered!